csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.shared.cs

```diff
@@ -408,6 +408,33 @@ public void AppendExecutionProvider(string providerName, Dictionary<string, stri
             var appender = new ExecutionProviderAppender(utf8ProviderName);
             ProviderOptionsUpdater.Update(providerOptions, handle, appender.Appender);
         }
+
+        public void AppendExecutionProvider(OrtEnv env, IReadOnlyList<OrtEpDevice> epDevices, 
+                                            OrtKeyValuePairs epOptions)
```

Comment:
This could alternatively take IReadOnlyDictionary<string, string> and create the OrtKeyValuePairs internally before calling OrtSessionOptionsAppendExecutionProvider_V2. That might be a little more user friendly.